### PR TITLE
Add tep validation and tep creation tools

### PR DESCRIPTION
**Pull request by afrittoli**
Add a tep validate helper: split the script into two helpers:
- table that generates the TEP table (this will be used by user and for
  CI to check that the table is up to date)
- validate that checks if TEPs are well formed (this will be used for CI)

Add a tep creation tool which allows to create a new TEP from a
template based on title and authors. Number, date and status are
set automatically.

Add a docker file for the teps tool.
Update the TEP table. 
Fix validation failures:
```
WARNING:root:[InvalidTep('TEP filenames should match /^[0-9]4-/. Found: XXXX-workspace-paths.md'), InvalidTep('No TEP number title (# TEP-NNNN) in /System/Volumes/Data/go/src/github.com/tektoncd/community/teps/XXXX-workspace-paths.md')]
WARNING:root:[InvalidTep('TEP number TEP-0028 from filename does not match TEP number TEP-0027 from title (# TEP-NNNN) in /System/Volumes/Data/go/src/github.com/tektoncd/community/teps/0028-task-execution-status-at-runtime.md')]
WARNING:root:[InvalidTep('TEP filenames should match /^[0-9]4-/. Found: XXXX-step-workspaces.md'), InvalidTep('No TEP number title (# TEP-NNNN) in /System/Volumes/Data/go/src/github.com/tektoncd/community/teps/XXXX-step-workspaces.md')]
WARNING:root:[InvalidTep('No TEP number title (# TEP-NNNN) in /System/Volumes/Data/go/src/github.com/tektoncd/community/teps/0022-trigger-immutable-input.md')]
```

Signed-off-by: Andrea Frittoli <andrea.frittoli@gmail.com>

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
/meow

File: teps/0029-step-workspaces.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ last-updated: 2020-10-02
 status: proposed
 ---
 
-# TEP-XXXX: Step and Sidecar Workspaces
+# TEP-0029: Step and Sidecar Workspaces
```
Comment (sbwsg) [inline teps/0029-step-workspaces.md:10]: 🎉

File: teps/0030-workspace-paths.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ creation-date: 2020-10-07
 last-updated: 2020-10-18
 status: proposed
 ---
-# TEP-XXXX: Workspace Paths
+# TEP-0030: Workspace Paths
```
Comment (sbwsg) [inline teps/0030-workspace-paths.md:9]: 🎉

**Pull request merged**